runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3105,13 +3106,24 @@ remoteCompile(J9VMThread *vmThread, TR::Compilation *compiler, TR_ResolvedMethod
    TR::PersistentInfo *persistentInfo = compInfo->getPersistentInfo();
    bool useAotCompilation = entry->_useAotCompilation;
 
-   bool aotCacheStore = useAotCompilation &&
-                        persistentInfo->getJITServerUseAOTCache() &&
-                        compInfo->methodCanBeJITServerAOTCacheStored(compiler->signature(), compilee->convertToMethod()->methodType());
-   bool aotCacheLoad = persistentInfo->getJITServerUseAOTCache() &&
-                       !TR::CompilationInfo::isCompiled(method) &&
-                       !entry->_doNotLoadFromJITServerAOTCache &&
-                       compInfo->methodCanBeJITServerAOTCacheLoaded(compiler->signature(), compilee->convertToMethod()->methodType());
+   bool aotCacheStore = false;
+   bool aotCacheLoad = false;
+   if (persistentInfo->getJITServerAOTCacheIgnoreLocalSCC())
+      {
+      aotCacheStore = entry->_useAotCacheCompilation &&
+                      compInfo->methodCanBeJITServerAOTCacheStored(compiler->signature(), compilee->convertToMethod()->methodType());
+      aotCacheLoad = persistentInfo->getJITServerUseAOTCache() &&
+                     !TR::CompilationInfo::isCompiled(method) &&
+                     !entry->_doNotLoadFromJITServerAOTCache &&
+                     compInfo->methodCanBeJITServerAOTCacheLoaded(compiler->signature(), compilee->convertToMethod()->methodType());
+      }
+   else
+      {
+      aotCacheStore = useAotCompilation && persistentInfo->getJITServerUseAOTCache() &&
+                      compInfo->methodCanBeJITServerAOTCacheStored(compiler->signature(), compilee->convertToMethod()->methodType());
+      aotCacheLoad = aotCacheStore && !entry->_doNotLoadFromJITServerAOTCache &&
```

Comment:
Previously, AOT cache loads could be done only on interpreted methods. Here the check on `!TR::CompilationInfo::isCompiled(method)` has disappeared. Was this done on purpose?

Reply:
Sorry, this is an artifact of the commit structure - if you look at the diff overall there wasn't a `!TR::CompilationInfo::isCompiled(method)` test there to begin with. (I added it in the "change AOT load criteria commit"). A method not being compiled is a prerequisite for  `entry->_useAotCompilation`, so it would be redundant to add when `getJITServerAOTCacheIgnoreLocalSCC()` is false.

It does need to be added for determining `aotCacheLoad` when `getJITServerAOTCacheIgnoreLocalSCC()` is true, though. I took the criteria from the existing ones for AOT loads in `precompilationTasks()` here:

https://github.com/eclipse-openj9/openj9/blob/6a6d7afd73759f7bf4c786055bcb02bb35343030/runtime/compiler/control/CompilationThread.cpp#L7707-L7712

omitting the ones that didn't seem relevant. Though, now that I look at that code again I see that there's a `!_jitConfig->inlineFieldWatches` component that I might want to add, because I put that in the eligibility criteria for AOT cache stores.